### PR TITLE
Mark Task.IsFaulted as ensuring that Task.Exception is non-null

### PR DESCRIPTION
**Pull request by jasonmalinowski**

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs [hunk 1]
```diff
@@ -1523,6 +1524,7 @@ internal bool ExceptionRecorded
         /// <see cref="System.Threading.Tasks.TaskStatus.Faulted">TaskStatus.Faulted</see>, and its
         /// <see cref="Exception"/> property will be non-null.
         /// </remarks>
+        [MemberNotNullWhen(true, nameof(Exception))]
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:1527]: For this to impact external consumers, you'll also need to update the ref assembly.

Comment (jasonmalinowski) [reply src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:1527]: @stephentoub Are there are instructions I can follow?

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:1527]: Just make the same change (except using fully-qualified names) to the corresponding member in https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime/ref/System.Runtime.cs

Comment (jasonmalinowski) [reply src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:1527]: @stephentoub Updated.

File: src/libraries/System.Runtime/ref/System.Runtime.cs [hunk 2]
```diff
@@ -14505,6 +14505,7 @@ public partial class Task : System.IAsyncResult, System.IDisposable
         public bool IsCanceled { get { throw null; } }
         public bool IsCompleted { get { throw null; } }
         public bool IsCompletedSuccessfully { get { throw null; } }
+        [System.Diagnostics.CodeAnalysis.MemberNotNullWhen(true, nameof(Exception))]
```
Comment (stephentoub) [inline src/libraries/System.Runtime/ref/System.Runtime.cs:14508]:
```suggestion
        [System.Diagnostics.CodeAnalysis.MemberNotNullWhenAttribute(true, nameof(System.Exception))]
```

**Review by stephentoub — APPROVED** [review]

**Comment by jasonmalinowski** [comment]
@stephentoub Thanks!

**Pull request merged**